src/main/java/com/starbucks/controllers/RestControllers.java

```diff
@@ -19,30 +19,25 @@ public RestControllers(OrderService orderService) {
     }
 
     @PostMapping()
-    public String saveOrder(@RequestBody Order order) {
-        orderService.saveOrder ( order );
-        return order.toString ( );
+    public Order saveOrder(@RequestBody Order order) {
+        return orderService.saveOrder(order);
     }
 
     @PutMapping("/{id}")
-    public String updateOrder(@RequestBody Order order , @PathVariable("id") UUID id) {
-
-        try {
-            orderService.existOrder ( id );
-        } catch (RuntimeException ex) {
-            throw new ResourceAccessException ( "Order does not exist " );
-        }
-        orderService.updateOrder ( order , id );
-        return order.toString ( );
+    public Order updateOrder(@RequestBody Order order, @PathVariable("id") UUID id) {
+        orderService.existOrder(id);
+        return orderService.updateOrder(id, order);
     }
 
     @DeleteMapping("/{id}")
     public void deleteOrder(@PathVariable("id") UUID id) {
-        orderService.deleteOrdersById ( id );
+        orderService.existOrder(id);
```

Comment:
the same, only one call to service, controller shouldn't be involved in any business logic. And check if something is present in DB is a part of the business logic.